includes/Modules/Tag_Manager.php

```diff
@@ -498,35 +498,23 @@ protected function create_data_request( Data_Request $data ) {
 					return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'accountID' ), array( 'status' => 400 ) );
 				}
 
-				$usage_context = $data['usageContext'] ?: self::USAGE_CONTEXT_WEB;
-
-				if ( self::USAGE_CONTEXT_WEB === $usage_context && ! isset( $data['containerID'] ) ) {
-					/* translators: %s: Missing parameter name */
-					return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'containerID' ), array( 'status' => 400 ) );
-				}
-				if ( self::USAGE_CONTEXT_AMP === $usage_context && ! isset( $data['ampContainerID'] ) ) {
-					/* translators: %s: Missing parameter name */
-					return new WP_Error( 'missing_required_param', sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'ampContainerID' ), array( 'status' => 400 ) );
-				}
```

Comment:
Why was this removed? We should still catch these errors I think.
* If secondary AMP, require both.
* If primary AMP, require `ampContainerID`.
* If no AMP, require `containerID`.

This should maybe depend on the passed `usageContext` instead of `Context` - what do you think?

Reply:
I think I removed it when removing the conditional in the `data.set()` call which used to be dynamic based on `containerKey`. Now it passes both in all cases. I don't think there's a need to over-complicate things on the JS side to prevent sending container IDs based on usage context.

I updated the datapoint handler to require the necessary parameters based on `usageContext` which is also now required.